zeppelin-zengine/src/main/java/org/apache/zeppelin/plugin/PluginManager.java

```diff
@@ -83,8 +83,13 @@ public NotebookRepo loadNotebookRepo(String notebookRepoClassName) throws IOExce
       return null;
     }
     URLClassLoader classLoader = new URLClassLoader(urls.toArray(new URL[0]));
-    Iterator<NotebookRepo> iter = ServiceLoader.load(NotebookRepo.class, classLoader).iterator();
-    NotebookRepo notebookRepo = iter.next();
+    NotebookRepo notebookRepo = null;
+    try {
+      notebookRepo = (NotebookRepo) (Class.forName(notebookRepoClassName, true, classLoader)).newInstance();
```

Comment:
Could you still load the notebookRepo via ServiceLoader and just verify the classname equal to  notebookRepoClassName ?

Reply:
@cvaliente thanks for this fix and for highlighting the issue. We also ran to this classpath loading issue and applying this change seems to have fixed it! Prior to this ServiceLoader seems to find VFSNotebookRepo instead of the desired GitNotebookRepo.

Reply:
hmm, @cvaliente You are right, ServiceLoader is not necessary here. I will remove ServiceLoader mechanism in another PR.